### ISSUE TITLE
[3.3.5] Phantom Guests (Karazhan) remain in sitting position in combat

### ISSUE THREAD

**Issue by Saben65** [TrinityCore/TrinityCore#16832]
**Description**:
[Phantom Guests](http://www.wowhead.com/npc=16409/phantom-guest) will remain in a sitting position when aggro'd. Additionally, when they reset, they will go back to their chairs and start sitting in the air slightly above the seat.

The ones that are bugged are the ones located in Moroe's room sitting at the chairs with `bytes1`=5 in `creature_addon`. The ones dancing before his room appear to be fine.

When Aggro'd they will be sitting in air.
![Phantom Guest Combat](http://i.imgur.com/laBa45I.jpg)

When reset they will float above their seats.
![Phantom Guest Reset](http://i.imgur.com/l6SfnCT.jpg)

**Branch(es)**: 335

**TC hash/commit**:
Rev: ecf15a156b07

**TDB version**:
TDB: 335.60

**Comment by Killyana** [comment]
This is a general issue, npcs must update emote_state once in combat.

**Comment by Rushor** [comment]
setting the emote_state to 0 in creature::entercombat sould fix this no?
is there any exception which should not update the emotestate one the fight begins?

**Comment by Killyana** [comment]
I don't know any npc that should keep a specific emote state once IC, Even if it's the case it must be handled via the script.

**Comment by Saben65** [comment]
Adding this seems to fix the solution for most of them still sitting while in combat. However, when they reset they still float above the seat.
 
```SQL
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(16409,0,8,0,0,0,100,0,0,0,0,0,17,0,0,0,0,0,0,0,0,0,0,0,0,0,0,"Phantom Guest - In Combat - Set Emote State 0");
```

**Comment by Killyana** [comment]
Yes but we need a generic solution like setting the emote_state to 0 in creature::entercombat

**Comment by Rushor** [comment]
we could add the change for the emote here:
https://github.com/TrinityCore/TrinityCore/blob/cd777f97a182c582221bf5194e4075d92c532679/src/server/game/Entities/Unit/Unit.cpp#L9912 (not 100% shure it's the right place)

**Comment by tkrokli** [comment]
According to https://trinitycore.atlassian.net/wiki/display/tc/Emotes there are 3 sit emotes for chairs.
From Emotes.dbc (3.3.5):
```
  id    field1                  field2  field3  field4  field5  field6
----    --------------------    ------  ------  ------  ------  ------
 415    STATE_SIT_CHAIR_MED       103       0       2       0       0
 426    STATE_SIT_CHAIR_HIGH      104       0       2       0       0
 461    STATE_SIT_CHAIR_LOW       102       0       2       0       0
```
Maybe the wrong emote is in use for the NPCs that don't sit correctly in their chairs?

**Comment by malcrom** [comment]
I like Rushors idea.

**Comment by Rushor** [comment]
not only emotestate, but also the bytes1 from creature_(template)_addon must be set to 0 once the creature enters combat. 

I don't know if there is any exception where creatures don't remove their emotestate/standstate after entering combat.

**Comment by Rushor** [comment]
hm try this:

```diff
diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index a0838e7..4454c9b 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -9250,8 +9250,9 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
         ToCreature()->SendAIReaction(AI_REACTION_HOSTILE);
         ToCreature()->CallAssistance();
 
-        // Remove emote state - will be restored on creature reset
+        // Remove emote state and bytes - will be restored on creature reset
         SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
+        SetUInt32Value(UNIT_FIELD_BYTES_1, UNIT_STAND_STATE_STAND);
     }
 
     // delay offhand weapon attack to next attack time
```

**Comment by ariel-** [comment]
Better use Unit::SetStandState

**Comment by Killyana** [comment]
This commit was made https://github.com/TrinityCore/TrinityCore/pull/17542 to fix this issue.

**Comment by Saben65** [comment]
I don't have a running server to test this atm. Was this fixed with the PR and can this be closed?

**Comment by tkrokli** [comment]
@Keader : could you please add a description for why this issue was reopened, like what kind of issue/problem still remains or if this issue was not fixed by PR #17542 (Core/Unit: remove emote state when creature attacks another unit) / commit https://github.com/TrinityCore/TrinityCore/commit/a732749d721eefe7558d23e44c5f8dc18194e6d4 ?

**Comment by Keader** [comment]
After 26b6422ded4c02c8f95fb8342e562474b4fa0cd4

> When reset they will float above their seats.

Is back...
Idk if was fixed before, but still present.
And stuck with sit position, i cant reproduce anymore.

(Commit reverted because is broken all creatures that use Hover).

**Comment by Aokromes** [comment]
this bug is still valid?

**Comment by Killyana** [comment]
Npc reset above their seat is an old issue not related to this.

**Comment by Rushor** [comment]
Is there some kind of position update wrong?

**Comment by Killyana** [comment]
No , if you check retail, npcs must ignore collusion with gob type chair: 
![image](https://user-images.githubusercontent.com/6587064/77598267-a1e4d380-6f01-11ea-9cb6-c976100b76c6.png)

**Comment by Aokromes** [comment]
> No , if you check retail, npcs must ignore collusion with gob type chair:
> ![image](https://user-images.githubusercontent.com/6587064/77598267-a1e4d380-6f01-11ea-9cb6-c976100b76c6.png)

on retail they changed interaction with gos on cata or something like that, for example, if you go gundrak on 3.3.5 and jump over the "altars" of the bosses you can stay over the activated state, on retail the same was possible, now not anymore.
